daml-assistant/src/DAML/Assistant/Env.hs

```diff
@@ -147,10 +199,9 @@ getDamlPath = wrapErr "Determining daml home directory." $ do
 -- environment variable.
 getProjectPath :: IO (Maybe ProjectPath)
 getProjectPath = wrapErr "Detecting daml project." $ do
-        pathM <- firstJustM id
-            [ lookupEnv projectPathEnvVar
-            , findM hasProjectConfig . ascendants =<< getCurrentDirectory
-            ]
+        pathM <- overrideWithEnvVarMaybe @SomeException projectPathEnvVar Right $ do
```

Comment:
Should this be `@Void` and we have an exception instance for void? Or a `NoException` void like type? I hate seeing SomeException in the code.